services/user/Chainmail/plugin/src/serde_structs.rs

```diff
@@ -0,0 +1,37 @@
+use psibase::TimePointSec;
+use serde::{de, Deserialize};
+use serde_aux::prelude::*;
+
+use crate::bindings::exports::chainmail::plugin::queries::Message;
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct TempMessageForDeserialization {
+    #[serde(deserialize_with = "deserialize_number_from_string")]
+    pub msg_id: u64,
+    pub receiver: String,
+    pub sender: String,
+    pub subject: String,
+    pub body: String,
+    #[serde(deserialize_with = "deserialize_timepoint")]
+    pub datetime: u32,
+}
```

Comment:
Hmm, is it possible to deserialize with either:
* The `SavedMessage` struct in the service, or
* The generated `Message` object from your bindings?

Just seems a bit strange to essentially have three definitions of a Message type... Maybe it's needed though?

Reply:
I agree it seems there should be a better way, and i'm pushing that direction, but I haven't found a good one yet. At least isolating these definitions as I hvae in this project starts to build the required output of a potential tweak to the service macro that could auto-generate local types, but I haven't found a good way yet.

I think it's also noteworthy when you say "three definitions", `SavedMessage` is a table record and could therefore significantly diverge from the FE of a message, so I'd argue is more generally only 2 distinct definitions. I'm trying to get those 2 to be the same. Haven't gotten there yet.